### ISSUE TITLE
sniprun.vim tries to index field 'fn' ('nil value')

### ISSUE THREAD

**Issue by lmartinez-mirror** [michaelb/sniprun#56]
**Describe the bug**
Sniprun fails to install properly, preventing it from loading at all.

**What do the bug concerns? Sniprun itself, a particular interpreter?**
The bug manifests in the Vim script parts of the plugin.

**To Reproduce**
Assuming Neovim 0.4.4 and vim-plug are already installed and set:
1. Install `sniprun` using the recommended method (`Plug 'michaelb/sniprun', {'do': 'bash install.sh'}`)
2. After the script runs, both vim-plug and Neovim will report this error:
```sh
Error detected while processing /home/luis/.local/share/nvim/plugged/sniprun/plugin/sniprun.vim:
line    1:
E5105: Error while calling lua chunk: .../luis/.local/share/nvim/plugged/sniprun//lua/sniprun.lua:4: attempt to index field 'fn' (a nil value)
```

**Expected behavior**
Neovim should install and load the program without raising any error, at the very least.

**Environment:**
 - Neovim version 0.4.4

**Additional context**
I am trying to package this for the AUR, however it isn't clear where the `sniprun` binary should be relative to the tree structure of the plugin. Some additional help here would be great, too!

**Comment by michaelb** [comment]
i'm seeing to this issue, I can guess where it comes from, and I hope it's not a neovim 0.4.4 / 0.5 incompatibility

The sniprun binary should be at  target/release/sniprun from the repository root, at least that's where the lua code expect it to be

**Comment by michaelb** [comment]
should be resolved by the PR #57 and sniprun v0.5.2 

Unfortunately, it is due by the use of some 0.4.x - incompatible API in the Lua code. 

I can't exactly use the old vimscript with newer versions of the plugin, since the transition to Lua led to a significant rewrite of the Rust part of the plugin that carry the most of the new features.

What does it mean? Effectively, neovim 0.4.x can't work with sniprun >= v0.5

I hacked a bit the install.sh script, so that it only installs the latest version on sniprun compatible with the neovim version, so in your case that should be v0.4.9

**Comment by lmartinez-mirror** [comment]
Sure thing, thank you! Closing for now and moving this discussion there.
